org/mozilla/jss/CryptoManager.c

```diff
@@ -120,27 +120,8 @@ int ConfigureOCSP(
      * strings associated with these args
      */
 
-    if (ocspResponderURL) {
-        ocspResponderURL_string =
-            (char*) (*env)->GetStringUTFChars(env, ocspResponderURL, NULL);
-        if (ocspResponderURL_string == NULL) {
-            JSS_throwMsg(env, GENERAL_SECURITY_EXCEPTION,
-                    "OCSP invalid URL");
-            result = SECFailure;
-            goto finish;
-        }
-    }
-
-    if (ocspResponderCertNickname) {
-        ocspResponderCertNickname_string =
-            (char*) (*env)->GetStringUTFChars(env, ocspResponderCertNickname, NULL);
-        if (ocspResponderCertNickname_string == NULL) {
-            JSS_throwMsg(env, GENERAL_SECURITY_EXCEPTION,
-                    "OCSP invalid nickname");
-            result = SECFailure;
-            goto finish;
-        }
-    }
+    ocspResponderURL_string = JSS_RefJString(env, ocspResponderURL);
```

Comment:
@edewata -- The behavior changes here because we no longer throw `"OCSP invalid URL"` in the event of decode failure. (Notice the loop change on new line 133 as well). 

This ties into question 1.

Reply:
The change in line 133 might actually hide the memory issue. If ocspResponderURL is not NULL but if ocspResponderURL_string is NULL, the code will be skipped as if the ocspResponderURL was NULL, so the behaviour might be wrong. This is probably a corner case and the code might fail in other places as well due to memory issue, but I'd suggest reverting to the original code for line 133.

Reply:
I've added error checking as part of referencing and updated the conditional on new line 172 to preserve original behavior. (OCSP responder is neither enabled nor disabled in the event of an error).